runtime/model-protobuf/src/main/java/io/aklivity/zilla/runtime/model/protobuf/internal/ProtobufParser.java

```diff
@@ -32,44 +36,55 @@ public final class ProtobufParser
 {
     private final FileDescriptor[] dependencies;
 
+    private static final Pattern SYNTAX_PATTERN = Pattern.compile(
+            "^\\s*syntax\\s*=\\s*[\"'](proto2|proto3)[\"']\\s*;",
+            Pattern.MULTILINE | Pattern.CASE_INSENSITIVE
+    );
+
+    private static final String PROTO2_SYNTAX = "proto2";
+    private static final String PROTO3_SYNTAX = "proto3";
+
+    public enum ProtoSyntax
+    {
+        PROTO2,
+        PROTO3,
+        UNKNOWN
+    }
```

Comment:
Add a `private static ProtoSyntax of(String syntax)` method that attempts to resolve the enum, returning `UNKNOWN` if not.  We'll use it later to simplify the code below.